### PR TITLE
Import and modernize bocoup/test262-regexp-generator

### PR DESCRIPTION
**Pull request by ptomato**
https://github.com/bocoup/test262-regexp-generator was used to generate the tests in `test/built-ins/RegExp/CharacterClassEscapes/`. It makes sense for this script to live in the same repo as the tests it generates. Especially since in the meantime, the tests themselves have gotten out of sync with the upstream.

This imports the script, modernizes and cleans up the code, implements the changes necessary to match the changes made in the meantime to the tests, and integrates the script into the build process.

### CODE REVIEW

**Comment by ptomato** [comment]
Unfortunately, `ci_build.sh` fails due to my changes to `make.py` but that does not result in the CI job failing. I'll sort that out later.

File: tools/regexp-generator/package.json
Comment (mathiasbynens) [inline tools/regexp-generator/package.json:5]: Since this is not meant to be published, we could add

```json
"private": true
```

and remove unnecessary fields:

```suggestion
  "private": true,
  "main": "index.mjs",
```

**Review by mathiasbynens — APPROVED** [review]
Thanks for your work on this!

File: tools/regexp-generator/package.json
Comment (ljharb) [reply tools/regexp-generator/package.json:5]: i believe name and version are always required, even for private packages

Comment (mathiasbynens) [reply tools/regexp-generator/package.json:5]: Nope, they’re only required when you want to publish the package, which isn’t the case here

File: tools/regexp-generator/index.mjs
Comment (Ms2ger) [inline tools/regexp-generator/index.mjs:81]: Indentation looks off in this commit (Implement downstream changes). Tabs v spaces?

**Review by Ms2ger — APPROVED** [review]
lgtm modulo other comments

**Comment by ptomato** [comment]
Rather than mess with the Circle-CI images to get one that has both Python and Node installed, I'll migrate the linter and test generation jobs to GitHub Actions where they can run on the `ubuntu-latest` image which has everything already. I'll do that in a separate PR, and hopefully make some improvements along the way. Let's consider this PR on hold for the moment.

**Comment by ptomato** [comment]
With CI set up to handle npm dependencies, I've rebased this and will land it now.

**Pull request merged**
